### PR TITLE
tests getWalletAccountByAddress, addsecretkeystowallet

### PR DESCRIPTION
**Pull request by julienbrs**
testing getWalletAccountByAddress, addSecretKeysToWallet and setBaseAccount

One test isn't successful for the moment, but only because refactor changes for testnet 23 (https://github.com/massalabs/massa-web3/pull/256) hasn't been merged yet

### CODE REVIEW

File: test/web3/walletClient.spec.ts [hunk 1]
```diff
@@ -83,4 +92,137 @@ describe.skip('WalletClient', () => {
...
+        await WalletClient.walletGenerateNewAccount(),
+      ];
+
+      await web3Client.wallet().addAccountsToWallet(accounts);
+
+      const targetAccount = accounts[1]; // Assume we want to find the second account
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      const fetchedAccount = web3Client
+        .wallet()
+        .getWalletAccountByAddress(targetAccount.address!);
```
Comment (gregLibert) [inline test/web3/walletClient.spec.ts:110]: Can you fix the linter error?

File: test/web3/walletClient.spec.ts [hunk 2]
```diff
@@ -83,4 +92,137 @@ describe.skip('WalletClient', () => {
...
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      const secretKeys: string[] = [receiverPrivateKey, receiverPrivateKey];
+
+      const addedAccounts = await web3Client
+        .wallet()
+        .addSecretKeysToWallet(secretKeys);
+      const walletAccounts = await web3Client.wallet().getWalletAccounts();
+      expect([baseAccount, addedAccounts[0]]).toStrictEqual(walletAccounts);
+      expect(addedAccounts.length).toBe(1); // only one unique account should be added
+      expect(web3Client.wallet().getWalletAccounts().length).toBe(2); // only one unique account should be added
```
Comment (gregLibert) [inline test/web3/walletClient.spec.ts:176]: Why is it 2 and not 1?

File: test/web3/walletClient.spec.ts [hunk 3]
```diff
@@ -53,6 +61,7 @@ describe.skip('WalletClient', () => {
     });
 
     test('should throw error if account is not valid', async () => {
+      await web3Client.wallet().cleanWallet();
```
[the remark this answers is not in the document]
Comment (julienbrs) [reply test/web3/walletClient.spec.ts:64]: cleanWallet is called at forEach to clean environmnent. Here cleanWallet is useless, will be removed

File: test/web3/walletClient.spec.ts (hunk 2 above)
Comment (julienbrs) [reply test/web3/walletClient.spec.ts:167]: Because there is also the baseAccount

**Review by gregLibert — CHANGES_REQUESTED** [review]
```ts
await web3Client.wallet().cleanWallet();
```
is still present, can you remove it please ?

**Review by gregLibert — APPROVED** [review]

**Comment by gregLibert** [comment]
@julienbrs can you resolv conflict ?

**Review by gregLibert — APPROVED** [review]

**Pull request merged**